src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala

```diff
@@ -255,12 +255,16 @@ private[internal] trait TypeConstraints {
 
         tvar.constr.inst = NoType // necessary because hibounds/lobounds may contain tvar
 
-        val newInst =
+        val newInst = try {
           if (up || tvar.constr.hiBounds.exists(isSingleType)) { // If we have a singleton upper bound then we should use it.
             if (depth.isAnyDepth) glb(tvar.constr.hiBounds) else glb(tvar.constr.hiBounds, depth)
           } else {
             if (depth.isAnyDepth) lub(tvar.constr.loBounds) else lub(tvar.constr.loBounds, depth)
           }
+        } catch { case unsolvable: NoCommonType =>
```

Comment:
Hmm, I'm concerned about the performance impact. Can/should we handle this by returning an error result? /cc @retronym